### ISSUE TITLE
axe-core fails when evaluating valid aria-* value as ID reference list (aria-valid-attr-value)

### ISSUE THREAD

**Issue by djalmaaraujo** [dequelabs/axe-core#4172]
### Product

axe-core

### Product Version

4.7

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
The `aria-valid-attr-value` rule should accept any kind of string because the "ID reference list" says that it can contain an Unconstrained value (string type).

`Invalid ARIA attribute value: aria-controls="radix-:rj:-content-wordpress"`

Deque University rule link: https://dequeuniversity.com/rules/axe/4.7/aria-valid-attr-value

Please correct me if I am wrong, but the error with `aria-controls` in this case is generic to all values. After reading https://w3c.github.io/aria/#valuetype_idref_list, I realized that the value for `aria-*` attributes could be many things, including a `string`, with the following description: Unconstrained value type.

Meaning: We can use `:` semicolon in the value, same as IDS, because it's valid.

I cannot testify that the value `radix-:rj:-xyzzsse` works perfectly in both VoiceOver/NVDA or JAWS, but at least from the DOM perspective, it's a valid value.

Can you reconsider accepting any kind of string, or explain why we cannot use some chars?

Reference: https://w3c.github.io/aria/#propcharacteristic_value

#### Actual
Axe warns that the `aria-valid-attr-value` is invalid for `radix-:rj:-trigger-wordpress`.

<img width="2500" alt="image" src="https://github.com/dequelabs/axe-core/assets/3402/e72f0e7a-31bb-48f3-8e5a-16ae06050dab">

#### How to Reproduce
<img width="2558" alt="image" src="https://github.com/dequelabs/axe-core/assets/3402/0a0d577a-9a11-496f-9f1e-ff6ff29eb666">

1. Open https://sms7cl.csb.app/
2. Run the axe DevTools with any version above 4.7 (my initial test version)
3. You should get a report for: https://dequeuniversity.com/rules/axe/4.7/aria-valid-attr-value?application=AxeChrome

#### Additional context

Some UI libraries use different chars to generate IDs, classes, etc. One is Radix, a library that tried to be as accessible as possible since day zero.

Issue reference: https://github.com/radix-ui/primitives/issues/1703

If there's a proper explanation for why we cannot use semicolons or other chars, it would be nice to know, and I could open an issue on Radix with the explanation so they can evaluate it.

The thing is that today if you use Radix (many companies are adopting it), you may have to disable `aria-valid-attr-value` from your axe-core runners, which will impact other things.

Thank you!

**Comment by straker** [comment]
Thanks for the issue. So in this case I believe the issue is not that the value is not a valid ID, but that the referenced ID does not exist on the page. When we check for valid values of aria attributes, if the value is an ID reference(s) we also validate that the element exists on the page by doing `document.getElementById(ref)`. If no element is returned we report a violation that the value is not valid.

This follows the definition that an [ID reference](https://w3c.github.io/aria/#valuetype_idref) must "Reference to the ID of another element _in the same document_" (emphasis mine)

**Comment by djalmaaraujo** [comment]
@straker I believe you are correct, I updated the code sandbox example and the issue was related to an async loading component. I am closing to avoid confusion to others.
